lib/jazzy/grouper.rb

```diff
@@ -69,7 +69,42 @@ def self.group_custom_categories(docs, doc_index)
         children.each.with_index { |child, i| child.nav_order = i }
         make_group(children, category['name'], '')
       end
-      [group.compact, docs]
+
+      globals_grouped_by_file = self.group_globals_by_file(docs, doc_index)
+      [group.compact, docs + globals_grouped_by_file]
+    end
+
+    # Group root level Constants and Functions by the file they're declared in.
+    def self.group_globals_by_file(docs, doc_index)
+      types_to_files_to_symbols = {}
+      # Group symbols by file and type.
+      doc_index.each do |index_item|
+        if index_item.parent_in_code.nil? && (index_item.type.name == "Constant" || index_item.type.name == "Function")
+          file_name = File.basename(index_item.file, ".*")
+          types_to_files_to_symbols[index_item.type.plural_name] ||= {}
+          types_to_files_to_symbols[index_item.type.plural_name][file_name] ||= []
+          types_to_files_to_symbols[index_item.type.plural_name][file_name] << docs.delete(index_item)
+        end
+      end
+
+      # Create `SourceDeclaration`s for each file for each type of symbols it holds.
+      types_to_file_groups = {}
+      types_to_files_to_symbols.each do |type_name, files|
+        types_to_file_groups[type_name] ||= []
+        files.each do |file, declarations|
+          file_group = make_group(declarations, file, '')
```

Comment:
I _think_ this group is the one that should have the abstract "The following thingies are available globally" (you could write "are provided by <file>" given that you're adding source filenames into the UI.

And the group below (line 103) is the one that should have the abstract "The following files provide thingies" or something?